### PR TITLE
Update FS-1053-span.md

### PR DESCRIPTION
**Pull request by TIHan**

### CODE REVIEW

File: FSharp-4.5/FS-1053-span.md [hunk 1]
```diff
@@ -264,21 +264,34 @@ let F1() =
...
+        if true then
+            &y
+        else
+            &z
+    ()
 ```
-This only applies when calling members, not arbitrary let-bound functions. See "Alternatives" below.
+`&y` will also give the same error.
+
+The rationale behind this is not just for soundness reasons, but for gaurantees in the optimizer that local values like this will not escape in such a manner.
```
Comment (dsyme) [inline FSharp-4.5/FS-1053-span.md]: Typo `gaurantees`

File: FSharp-4.5/FS-1053-span.md [hunk 2]
```diff
@@ -498,7 +511,7 @@ machinery to express read-only references.
 
 # Notes
 
-* Implicit address-of is not applied for parameters to let-bound functions.  For example:
+* Implicit address-of was intially implemented for member methods but was taken out due to scoping rationale. Currently it is **not implemented**. If it was done again, it wouldn't be applied for parameters to let-bound functions. For example:
```
Comment (dsyme) [inline FSharp-4.5/FS-1053-span.md]: @TIHan Could you reword this a bit and place under `Alternatives` to capture the discussion we just had, e.g. that this is part of the corresponding C# design, and the example where the scope of the F# tmp is narrow, and that we will seek to re-instate this in F# 5.0 especially if API ergonomics make it necessary? thanks

Comment (TIHan) [reply FSharp-4.5/FS-1053-span.md]: Yup will do.

File: FSharp-4.5/FS-1053-span.md [hunk 3]
```diff
@@ -264,21 +264,34 @@ let F1() =
     c.InstanceProperty <- today.AddDays(2.0)
 ```
 
+#### Scoping for byrefs
 
-#### Implicit address-of when calling members with parameter type `inref<'T>` 
+The general idea is that a let-bound value cannot have its reference escape the scope from which it was declared, effectively its visibility. For example:
```
Comment (cartermp) [inline FSharp-4.5/FS-1053-span.md]: What does "effectively its visibility" mean?

Comment (TIHan) [reply FSharp-4.5/FS-1053-span.md]: The visibility scope of the value.

Comment (cartermp) [reply FSharp-4.5/FS-1053-span.md]: Okay, I think this makes sense as-is.

**Pull request merged**